tests/qml/helper.cpp

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "../../src/mozillavpn.h"
+#include "../../src/settingsholder.h"
+#include "l18nstrings.h"
+#include "helper.h"
+
+TestHelper::TestHelper() {
+  FeatureList::instance()->initialize();
+  m_whatsNewModel = new WhatsNewModel();
+}
+
+TestHelper* TestHelper::instance() {
+  static TestHelper* s_instance = new TestHelper();
+  return s_instance;
+}
+
+bool TestHelper::mainWindowLoadedCalled() {
+  // I don't understand why I need to call instance()
+  // here, would appreciate a c++ lesson and/or correction
+  // of my code.
+  return instance()->m_mainWindowLoadedCalled;
```

Comment:
You have spot a bug in your code. You don't need to use `instance()->..` here. This method should be:
```
bool TestHelper::mainWindowLoadedCalled() const { return m_mainWindowLoadedCalled; }
```
(note the `const`).

The reason why you need `instance()` is the following:

1. `TestHelper` is called the first time in the main, when this line runs: `QUICK_TEST_MAIN_WITH_SETUP(qml_tests, TestHelper)`. Let's call this object "A"
2. This object is created but it's not set into `s_instance` because `::instance()` is not called.
3. Later on, in your code, you run `MozillaVPN::mainWindowLoaded`
4. ... in here you call `TestHelper::instance`
5. ... and here you run `new TestHelper`, creating a new instance of `TestHelper`. Let's call this "B".
6. `MozillaVPN::mainWindowLoaded` called `setMainWindowLoadedCalled` using "B".
7. But when `TestHelper::mainWindowLoadedCalled` is executed by the QML engine, "A" is used. And "A" returns false.

The fix I suggest is the following:
1. The `TestHelper` CTOR should set the `s_instance`: (ignore the indentation please)
```
namespace { TestHelper* s_instance = nullptr; }
TestHelper::TestHelper() {
  Q_ASSERT(s_instance == nullptr); // In this way we are 100% sure there is only 1 instance of TestHelper around.
  s_instance = this;
  // .. all the rest of the code
}
```

2. Unset the `s_instance` in the DTOR:
```
TestHelper::~TestHelper() {
  Q_ASSERT(s_instance == this); // If it's not us, what is this?
  s_instance = nullptr;
}
```

3. a simple `::instance()` static method:
```
TestHelper* TestHelper::instance() {
  Q_ASSERT(s_instance); // We _must_ have an object around.
  return s_instance;
}
```

(Without my silly comments)

Reply:
Discussed with baku; we're sticking with magic statics. 

We don't need to use the macro to run the tests, which means we don't need a public constructor, so everything will work!